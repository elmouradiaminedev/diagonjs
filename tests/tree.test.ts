import { expect, describe, it } from "vitest";
import Diagon, { TreeTranslationOptions, TreeTranslationStyle } from "../dist";

function generateAllOptions(): TreeTranslationOptions[] {
  const styles: TreeTranslationStyle[] = [
    "Unicode",
    "Unicode 2",
    "ASCII 1",
    "ASCII 2",
    "ASCII 3",
    "Unicode right top",
    "Unicode right center",
    "Unicode right bottom",
  ];

  const allOptions: TreeTranslationOptions[] = [];

  for (const style of styles) {
    allOptions.push({
      style,
    });
  }

  return allOptions;
}

describe("Tree expression translation", () => {
  generateAllOptions().forEach((options) => {
    describe(`With options ${JSON.stringify(options)}`, () => {
      it("should translate a simple tree", () => {
        expect(
          Diagon.translate.tree(
            "Linux\n  Android\n  Debian\n    Ubuntu\n      Lubuntu\n      Kubuntu\n      Xubuntu\n      Xubuntu\n    Mint\n  Centos\n  Fedora",
            options,
          ),
        ).toMatchSnapshot();
      });
    });
  });
});
