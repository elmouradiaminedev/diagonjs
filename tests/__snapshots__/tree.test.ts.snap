// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tree expression translation > With options {"style":"ASCII 1"} > should translate a simple tree 1`] = `
"Linux
 +-Android
 +-Debian
 |   +-Ubuntu
 |   |   +-Lubuntu
 |   |   +-Kubuntu
 |   |   +-Xubuntu
 |   |   \`-Xubuntu
 |   \`-Mint
 +-Centos
 \`-Fedora
"
`;

exports[`Tree expression translation > With options {"style":"ASCII 2"} > should translate a simple tree 1`] = `
"Linux
 +--Android
 +--Debian
 |   +--Ubuntu
 |   |   +--Lubuntu
 |   |   +--Kubuntu
 |   |   +--Xubuntu
 |   |   \`--Xubuntu
 |   \`--Mint
 +--Centos
 \`--Fedora
"
`;

exports[`Tree expression translation > With options {"style":"ASCII 3"} > should translate a simple tree 1`] = `
"Linux
 |--Android
 |--Debian
 |   |--Ubuntu
 |   |   |--Lubuntu
 |   |   |--Kubuntu
 |   |   |--Xubuntu
 |   |   \`--Xubuntu
 |   \`--Mint
 |--Centos
 \`--Fedora
"
`;

exports[`Tree expression translation > With options {"style":"Unicode 2"} > should translate a simple tree 1`] = `
"Linux
 ├──Android
 ├──Debian
 │   ├──Ubuntu
 │   │   ├──Lubuntu
 │   │   ├──Kubuntu
 │   │   ├──Xubuntu
 │   │   └──Xubuntu
 │   └──Mint
 ├──Centos
 └──Fedora
"
`;

exports[`Tree expression translation > With options {"style":"Unicode right bottom"} > should translate a simple tree 1`] = `
"Linux
 ├──Android
 ├──Debian
 │   ├──Ubuntu
 │   │   ├──Lubuntu
 │   │   ├──Kubuntu
 │   │   ├──Xubuntu
 │   │   └──Xubuntu
 │   └──Mint
 ├──Centos
 └──Fedora
"
`;

exports[`Tree expression translation > With options {"style":"Unicode right center"} > should translate a simple tree 1`] = `
"Linux
 ├──Android
 ├──Debian
 │   ├──Ubuntu
 │   │   ├──Lubuntu
 │   │   ├──Kubuntu
 │   │   ├──Xubuntu
 │   │   └──Xubuntu
 │   └──Mint
 ├──Centos
 └──Fedora
"
`;

exports[`Tree expression translation > With options {"style":"Unicode right top"} > should translate a simple tree 1`] = `
"Linux
 ├──Android
 ├──Debian
 │   ├──Ubuntu
 │   │   ├──Lubuntu
 │   │   ├──Kubuntu
 │   │   ├──Xubuntu
 │   │   └──Xubuntu
 │   └──Mint
 ├──Centos
 └──Fedora
"
`;

exports[`Tree expression translation > With options {"style":"Unicode"} > should translate a simple tree 1`] = `
"Linux
 ├──Android
 ├──Debian
 │   ├──Ubuntu
 │   │   ├──Lubuntu
 │   │   ├──Kubuntu
 │   │   ├──Xubuntu
 │   │   └──Xubuntu
 │   └──Mint
 ├──Centos
 └──Fedora
"
`;
